app/src/components/misc/PersonaSurvey/ModalFooter.tsx

```diff
@@ -1,88 +1,84 @@
+  const surveyLength = useMemo(() => Object.keys(SurveyConfig).length, []);
   const isSharedListUser = window.location.href.includes(PATHS.SHARED_LISTS.VIEWER.RELATIVE);
 
-  const disableContinue = () => {
-    if (currentPage === 0) return false;
-    if (userPersona[OptionsConfig[currentQuestionnaire as QuestionnaireType]?.key]?.length) return false;
+  const disableContinue = useMemo(() => {
+    if (currentPage === SurveyPage.GETTING_STARTED) return false;
+    if (userPersona[currentQuestionnaire as QuestionnaireType]?.length) return false;
     return true;
-  };
+  }, [currentPage, currentQuestionnaire, userPersona]);
 
   const handleMoveToNextPage = () => {
+    const index = Object.keys(SurveyConfig).indexOf(currentPage);
+
     switch (currentQuestionnaire) {
       case QuestionnaireType.PERSONA:
         trackPersonaQ1Completed(userPersona.persona);
         submitAttrUtil(APP_CONSTANTS.GA_EVENTS.ATTR.PERSONA, userPersona.persona);
-        if (isSurveyModal) {
-          if (isSharedListUser || appMode === GLOBAL_CONSTANTS.APP_MODES.DESKTOP) {
-            //don’t show recommendation screen for shared list users or desktop users
-            dispatch(actions.updateIsPersonaSurveyCompleted(true));
-            return;
-          }
-          dispatch(actions.toggleActiveModal({ modalName: "personaSurveyModal", newValue: false }));
-          navigate(PATHS.GETTING_STARTED, {
-            replace: true,
-            state: {
-              src: "persona_survey_modal",
-              redirectTo: window.location.pathname,
-            },
-          });
-        }
         break;
     }
-    if (isSurveyModal || currentPage !== SurveyConfig.length - 1) {
-      dispatch(actions.updatePersonaSurveyPage(currentPage + 1));
+    if (isSurveyModal || index !== surveyLength - 1) {
+      handleSurveyNavigation(currentPage, dispatch);
+
+      if (isSurveyModal && index === surveyLength - 1) {
+        if (isSharedListUser || appMode === GLOBAL_CONSTANTS.APP_MODES.DESKTOP) {
+          //don’t show recommendation screen for shared list users or desktop users
+          dispatch(actions.updateIsPersonaSurveyCompleted(true));
+          return;
+        }
+        dispatch(actions.toggleActiveModal({ modalName: "personaSurveyModal", newValue: false }));
+      }
     } else {
       callback?.();
     }
   };
 
   return (
     <>
-      <div className="survey-footer w-full">
+      <div className="survey-footer w-full rq-modal-footer">
         <Row justify="space-between" align="middle">
           <Col>
-            {isSurveyModal && currentPage === 0 ? (
+            {isSurveyModal && currentPage === SurveyPage.GETTING_STARTED ? (
               <>
                 <span className="survey-modal-emoji">😁</span> We are excited to see you here
               </>
             ) : (
               <>
-                {currentPage} / {SurveyConfig?.length - 1}
+                {Object.keys(SurveyConfig).indexOf(currentPage)} / {surveyLength - 1}
```

Comment:
can we get the `index` for currentPage only once at top
```suggestion
const currentPageIndex = useMemo(() => Object.keys(SurveyConfig).indexOf(currentPage), [currentPage])
```